internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -229,16 +230,21 @@ func resourceDataFactoryDatasetBinary() *pluginsdk.Resource {
 						},
 						// SFTP Specific field
 						"type": {
-							Type:     pluginsdk.TypeString,
-							Required: true,
+							Type:             pluginsdk.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
 							ValidateFunc: validation.StringInSlice([]string{
-								string(datafactory.TypeBasicDatasetCompressionTypeBZip2),
-								string(datafactory.TypeBasicDatasetCompressionTypeDeflate),
-								string(datafactory.TypeBasicDatasetCompressionTypeGZip),
-								string(datafactory.TypeBasicDatasetCompressionTypeTar),
-								string(datafactory.TypeBasicDatasetCompressionTypeTarGZip),
-								string(datafactory.TypeBasicDatasetCompressionTypeZipDeflate),
-							}, false),
+								string(datafactory.CompressionCodecBzip2),
+								string(datafactory.CompressionCodecDeflate),
+								string(datafactory.CompressionCodecGzip),
+								string(datafactory.CompressionCodecLz4),
+								string(datafactory.CompressionCodecLzo),
+								string(datafactory.CompressionCodecNone),
+								string(datafactory.CompressionCodecSnappy),
+								string(datafactory.CompressionCodecTar),
+								string(datafactory.CompressionCodecTarGZip),
+								string(datafactory.CompressionCodecZipDeflate),
+							}, true),
```

Comment:
(as above) this should be case sensitive here

```suggestion
							}, false),
```

Reply:
Thanks @tombuildsstuff for the suggestion! Data factory removed the sub-classes of the dataset compression type as listed below and this led to the missing of the enums:
```
Removed:
const (
	TypeBasicDatasetCompressionTypeBZip2 TypeBasicDatasetCompression = "BZip2"
	TypeBasicDatasetCompressionTypeDatasetCompression TypeBasicDatasetCompression = "DatasetCompression"
	TypeBasicDatasetCompressionTypeDeflate TypeBasicDatasetCompression = "Deflate"
	TypeBasicDatasetCompressionTypeGZip TypeBasicDatasetCompression = "GZip"
	TypeBasicDatasetCompressionTypeTar TypeBasicDatasetCompression = "Tar"
	TypeBasicDatasetCompressionTypeTarGZip TypeBasicDatasetCompression = "TarGZip"
	TypeBasicDatasetCompressionTypeZipDeflate TypeBasicDatasetCompression = "ZipDeflate"
)
```
Considering the actual supported value is the same, so I chose to use below enums:
```
const (
	CompressionCodecBzip2 CompressionCodec = "bzip2"
	CompressionCodecDeflate CompressionCodec = "deflate"
	CompressionCodecGzip CompressionCodec = "gzip"
	CompressionCodecLz4 CompressionCodec = "lz4"
	CompressionCodecLzo CompressionCodec = "lzo"
	CompressionCodecNone CompressionCodec = "none"
	CompressionCodecSnappy CompressionCodec = "snappy"
	CompressionCodecTar CompressionCodec = "tar"
	CompressionCodecTarGZip CompressionCodec = "tarGZip"
	CompressionCodecZipDeflate CompressionCodec = "zipDeflate"
)
```

hope it make sense to you, feel free to let me know if you have any questions

Reply:
@xiaxyi fwiw Golang uses Constants - not Enums, they work differently.

In this case can we fix the casing in those constants to be consistent with the original values - recasing these causes the issues highlighted above

Reply:
I think we can do that, at lease this doesn't affects the case sensitive scenario. I'll modify it and run acctest. besides this, do you ok with the expand/flatten part?